test/sqllogictest/transform/join_index.slt

```diff
@@ -385,10 +385,9 @@ Explained Query:
   Project (#1, #3, #5) // { arity: 3 }
     Join on=(#0 = #2 AND #4 = (#0 + 4)) type=differential // { arity: 6 }
       implementation
-        %2:baz[#0] » %0:foo[(#0 + 4)]KA » %1:bar[#0]KA
-      ArrangeBy keys=[[(#0 + 4)]] // { arity: 2 }
-        Filter (#0) IS NOT NULL // { arity: 2 }
-          Get materialize.public.foo // { arity: 2 }
+        %0:foo[#0]KA » %1:bar[#0]KA » %2:baz[#0]KA
+      ArrangeBy keys=[[#0]] // { arity: 2 }
```

Comment:
This plan looks sufficiently better than we might want to look in to the comment above and update it:

> # materialize#8002: it would be nice if this join used the indexes on foo(a)
> # and bar(a+4)

Reply:
Also, I'm a little unclear on how we did not change the dependence on `foo_idx`. Do we report "using" an index even if we are not using its indexed form? If that is right, perhaps "used indexes" is not as meaningful as it might otherwise be (it does not answer the question "am I used `foo` by way of an index").